src/torrent.cpp

```diff
@@ -1520,54 +1562,25 @@ bool is_downloading_state(int const st)
 		// the verification function verifies the distinguished name
 		// of a peer certificate to make sure it matches the info-hash
 		// of the torrent, or that it's a "star-cert"
-		ctx->set_verify_callback(std::bind(&torrent::verify_peer_cert, this, _1, _2), ec);
+		ctx->set_verify_callback(
+				std::bind(&torrent::verify_peer_cert, this, m_torrent_file->name(), _1, _2)
```

Comment:
technically, since `verify_peer_cert` is still a member function now, `m_torrent_file->name()` doesn't need to be passed along here